dio/lib/src/dio_event_processor.dart

```diff
@@ -89,6 +89,24 @@ class DioEventProcessor implements EventProcessor {
       headers: _options.sendDefaultPii ? headers : null,
       bodySize: dioError.response?.data?.length as int?,
       statusCode: response?.statusCode,
+      data: _getResponseData(dioError.response?.data),
     );
   }
+
+  /// Returns the response data, if possible according to the users settings.
+  Object? _getResponseData(dynamic data) {
+    if (!_options.sendDefaultPii) {
+      return null;
+    }
+    if (data is String) {
+      if (_options.maxResponseBodySize.shouldAddBody(data.codeUnits.length)) {
+        return data;
+      }
+    } else if (data is List<int>) {
```

Comment:
I wonder if there are more possible types here.

Reply:
Good point, I'll checking the documentation and test more cases if needed.